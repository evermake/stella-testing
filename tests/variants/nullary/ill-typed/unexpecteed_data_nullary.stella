language core;

extend with #variants, #nullary-variant-labels;

fn foo(bval : Bool) -> <| nval, sval |> {
	return <| nval = 0 |>
}

fn main(succeed : Bool) -> Nat {
  return succ(0)
}
