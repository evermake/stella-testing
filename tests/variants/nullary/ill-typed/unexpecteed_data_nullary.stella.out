An error occurred during typechecking!
===================
Primary type error:
===================
Type Error Tag: [ERROR_UNEXPECTED_DATA_FOR_NULLARY_LABEL]
expected a nullary label nval
========================
Alternative type errors:
========================
Type Error Tag: [ERROR_AMBIGUOUS_VARIANT_TYPE]
type inference for variants is not supported (use type ascriptions)